internal/rbd/rbd_util.go

```diff
@@ -481,15 +467,14 @@ func genSnapFromSnapID(ctx context.Context, rbdSnap *rbdSnapshot, snapshotID str
 			return err
 		}
 	}
-
-	err = updateSnapWithImageInfo(ctx, rbdSnap, cr)
-
-	return err
+	return nil
 }
 
 // genVolFromVolID generates a rbdVolume structure from the provided identifier, updating
 // the structure with elements from on-disk image metadata as well
-func genVolFromVolID(ctx context.Context, volumeID string, cr *util.Credentials, secrets map[string]string) (*rbdVolume, error) {
+// cleanupVol will be set during the DeleteVolume operation to delete the image,
```

Comment:
this adds more complexity to the function. is it not easier to understand if the DeleteVolume operation takes care of it? A better maintainable solution has my preference.

Reply:
no other rbdvolume related information is passed from genVolFromVolID to delete the image, in case of failure we return nil and error

Reply:
hmm, I don't like it. It should probably get cleaned up at one point.

Reply:
This is fixed now.